src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+            var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            if (container is null)
+            {
+                return;
+            }
+
+            var allMembers = container.GetMembers();
+            var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                         where
+                                             m.IsConversion() && // MethodKind.Conversion
+                                             m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                             container.Equals(m.Parameters[0].Type) // Convert from container type to other type
+                                         let typeName = m.ReturnType.ToMinimalDisplayString(semanticModel, position)
+                                         select SymbolCompletionItem.CreateWithSymbolId(
+                                             displayText: $"({typeName})", // The type to convert to
+                                             filterText: typeName,
+                                             sortText: $"{SortingPrefix}{typeName}",
+                                             symbols: ImmutableList.Create(m),
+                                             rules: CompletionItemRules.Default,
+                                             contextPosition: position,
+                                             properties: CreateCompletionHandlerProperty(CompletionHandlerConversion, (MinimalTypeNamePropertyName, typeName)));
+            context.AddItems(allExplicitConversions);
+
+            var indexers = allMembers.OfType<IPropertySymbol>().Where(p => p.IsIndexer).ToImmutableList();
+            if (!indexers.IsEmpty)
+            {
+                var indexerCompletion = SymbolCompletionItem.CreateWithSymbolId(
+                               displayText: $"this[]",
+                               filterText: "[",
+                               sortText: $"{SortingPrefix}",
+                               symbols: indexers,
+                               rules: CompletionItemRules.Default,
+                               contextPosition: position,
+                               properties: CreateCompletionHandlerProperty(CompletionHandlerIndexer));
+                context.AddItem(indexerCompletion);
+            }
+        }
+
+        private static ExpressionSyntax? GetParentExpressionOfInvocation(SyntaxToken token)
+        {
+            var syntaxNode = token.Parent;
+            return syntaxNode switch
+            {
+                MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                MemberBindingExpressionSyntax memberBinding => memberBinding.GetParentConditionalAccessExpression()?.Expression,
+                _ => null,
+            };
```

Comment:
afaict, this method has nothing to do with invocations.

Reply:
I change the name and added some doc comments.